schema.graphql

```diff
@@ -120,3 +302,285 @@ type GenesisFile @entity {
   id: ID!
   raw: String!
 }
+
+type MsgStakeSupplier @entity {
+  id: ID!
+  signer: Account!
+  supplier: Supplier!
+  owner: Account!
+  # we are not using Coin because we want to be able to query and sum by amount
+  stakeAmount: BigInt!
+  stakeDenom: String!
+  block: Block!
+  transaction: Transaction!
+  message: Message!
+}
+
+type MsgStakeSupplierService @entity {
+  id: ID!
+  stakeMsg: MsgStakeSupplier!
+  service: Service!
+  revShare: [SupplierRevShare]!
+  endpoints: [SupplierEndpoint]!
+}
+
+type MsgUnstakeSupplier @entity {
+  id: ID!
+  signer: Account!
+  supplier: Supplier!
+  block: Block!
+  transaction: Transaction!
+  message: Message!
+}
+
+type Supplier @entity {
+  # The ID is the operator address
+  id: ID!
+  operator: Account!
+  owner: Account!
+  # we are not using Coin because we want to be able to query and sum by amount
+  stakeAmount: BigInt!
+  stakeDenom: String!
+  stakeMsgs: [MsgStakeSupplier] @derivedFrom(field: "supplier")
+  unstakeMsgs: [MsgUnstakeSupplier] @derivedFrom(field: "supplier")
+  unstakingBeginEvents: [EventSupplierUnbondingBegin] @derivedFrom(field: "supplier")
+  unstakingEndEvents: [EventSupplierUnbondingEnd] @derivedFrom(field: "supplier")
+  # 0 = Staked, 1 = Unstaking, 2 = Unstaked
+  status: Int!
+  # Block where MsgUnstakeSupplier was received
+  unstakingBeginBlock: Block
+  # Height of the block where the supplier is going to be unstaked/unbonded.
+  # This is received with the EventSupplierUnbondingBegin event.
+  unstakingEndHeight: BigInt
+  # Block where the supplier was unstaked/unbonded.
+  # This is received with the EventSupplierUnbondingEnd event.
+  unstakingEndBlock: Block
+  services: [SupplierServiceConfig] @derivedFrom(field: "supplier")
```

Comment:
I acknowledge that the `Supplier` protobuf type calls this `services` but I think this is the rare exception where we should deviate in service of improving the downstream UX.

```suggestion
  serviceConfigs: [SupplierServiceConfig] @derivedFrom(field: "supplier")
```